core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/RawlsGroupComponent.scala

```diff
@@ -84,6 +84,19 @@ trait RawlsGroupComponent {
       userDeletes andThen userInserts andThen findGroupByName(rawlsGroupRef.groupName.value).map(_.updatedDate).update(Option(nowTimestamp))
     }
 
+    def overwriteGroupUsers(groupsWithUsers: Set[(RawlsGroupRef, Set[RawlsUserRef])]) = {
+      val groupNames = groupsWithUsers.map { case (groupRef, _) => groupRef.groupName.value }
+      val userDeletes = findUsersByGroupNames(groupNames).delete
+
+      val groupUserRecords = for {
+        (groupRef, users) <- groupsWithUsers
+        user <- users
+      } yield marshalGroupUsers(user, groupRef)
+
+      val userInserts = groupUsersQuery ++= groupUserRecords
+      userDeletes andThen userInserts andThen findGroupsByNames(groupNames).map(_.updatedDate).update(Option(nowTimestamp))
```

Comment:
explain plan shows that the delete hits the FK_GROUP_USERS_GROUP index and the update hits the PK index